pkg/controllers/utils/client.go

```diff
@@ -188,17 +203,19 @@ func FindOverlayNetworkNetID(client client.Reader) (*int32, error) {
 	return nil, fmt.Errorf("no overlay network found")
 }
 
-func DetectNetworkAttachmentOfNode(client client.Reader, node *corev1.Node) (underlayAttached, overlayAttached bool, err error) {
+func DetectNetworkAttachmentOfNode(client client.Reader, node *corev1.Node) (underlayAttached,
```

Comment:
How about creating another function `detectBGPNetworkAttachementOfNode`?
Also `FindUnderlayNetworkForNode` won't work well if one node has vlan and bgp networks at same time.

Reply:
If `FindUnderlayNetworkForNode` cannot work well, maybe creating another `FindBGPUnderlayNetworkForNode` is ok but not another `detectBGPNetworkAttachementOfNode`?

Reply:
yes, just what I want, an independent function for BGP network attachment detection

Reply:
Also we should take "one node belong to vlan and bgp at same time" into account, this will make bgp attachment detection not accurate.

Reply:
Done.